synapse/handlers/sync.py

```diff
@@ -1377,7 +1377,7 @@ async def generate_sync_result(
             one_time_key_counts = await self.store.count_e2e_one_time_keys(
                 user_id, device_id
             )
-            unused_fallback_key_types = (
+            unused_fallback_key_types = list(
                 await self.store.get_e2e_unused_fallback_key_types(user_id, device_id)
             )
```

Comment:
This goes straight into the `SyncResult` and is pretty much only going to be JSON-serialized from here on.
Having to take a copy to make mypy happy is unfortunate.

Reply:
At least it's a shallow copy :shrug:

Reply:
I'd be curious of the error here? Is something in *our* code unhappy or is it a type hint of like `json.dump`?

I think it'd be OK to add a `cast` with a comment explaining it is for performance.

Reply:
It's a type hint problem. Only `List`s and `Tuple`s are JSON-serializable, except they're mutable types. I would have updated the definition of `SyncResult` otherwise.